crates/sui-node/src/lib.rs

```diff
@@ -2065,7 +2065,15 @@ pub async fn build_http_server(
                     reverse_registry_id,
                 )
             } else {
-                sui_json_rpc::name_service::NameServiceConfig::default()
+                match CHAIN_IDENTIFIER
```

Comment:
Are we certain that `CHAIN_IDENTIFIER` is always initialized? Should we consider defaulting to unknown here? (to avoid a potential crash loop if not initialized)

Reply:
fwiw its explicitly set before this call, so it would be a bug if it wasn't initialized earlier.